app/src/common/shared/org/mozilla/vrbrowser/browser/engine/Session.java

```diff
@@ -541,7 +541,7 @@ void recreateSession() {
 
         mState = mState.recreate();
 
-        restore();
+        restore(false);
```

Comment:
I did not want a boolean instead of the flags. I was trying to not have to add any parameters to restore. Booleans as a function parameter are almost always confusing.

Reply:
The problem is that `restore` is called from `recreateSession`and `setActive`. `recreateSession` never needs to add the session but when called from `setActive` it depends on if the session has already been added or not so we can't rely on a single boolean to handle all the three cases.

Reply:
If the boolean can't represent the current state we need to figure out a different method. I'm still not sure I understand why it would not be correct. Seems like a bug to me.

Reply:
I have updated and:
- Moved the `onSessionAdd` code to an `createSession` method so we have all the session add/remove/link/unlink code in the Session unless until we do the refactor.
- Removed the session add tracking flag by removing the session from the `BrowserStore` on suspend. I think this works as we don't really want suspended sessions there so I think is a legit point of session removal from the BrowserStore.

I've tested all the scenarios and seems to stay in sync.

**Logcat** filtering by `SessionStore` will show the `BrowserStore`-`SessionStore` sync state including the alive vs suspended totals.

**Note**: Suspended session will be alive in the `SessionStore` but not in the `BrowserStore` now, we remove them from the `BrowserStore` when suspended, we only want alive ones there.